react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/FontAssetManager.kt

```diff
@@ -0,0 +1,68 @@
+package com.revenuecat.purchases.react.ui
+
+import android.content.res.AssetManager
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+
+
+internal object FontAssetManager {
+    @get:Synchronized
+    private var fontFamilyCache = mapOf<String, FontFamily>()
+    private val FILE_EXTENSIONS = arrayOf(".ttf", ".otf")
+
+    private const val FONT_PATH = "fonts/"
+
+    private enum class FontStyleExtension(
+        val extension: String,
+        val weight: FontWeight,
+        val style: FontStyle
+    ) {
+        REGULAR("", FontWeight.Normal, FontStyle.Normal),
+        BOLD("_bold", FontWeight.Bold, FontStyle.Normal),
+        ITALIC("_italic", FontWeight.Normal, FontStyle.Italic),
+        BOLD_ITALIC("_bold_italic", FontWeight.Bold, FontStyle.Italic);
+
+        companion object {
+            fun possibleFileNames(fontFamilyName: String) = values().flatMap { styleExtension ->
+                FILE_EXTENSIONS.map { fileNameExtension ->
+                    "$fontFamilyName${styleExtension.extension}$fileNameExtension"
+                }
+            }
+        }
+    }
+
+    @Synchronized
+    fun getFontFamily(fontFamilyName: String, assetManager: AssetManager): FontFamily? {
+        val cachedFontFamily = fontFamilyCache[fontFamilyName]
+        if (cachedFontFamily != null) {
+            return cachedFontFamily
+        }
+        val existingFontFileNames = assetManager.list(FONT_PATH)?.toList() ?: emptyList()
```

Comment:
Note that I moved this outside the inner loop, since we only really need to do this once.